src/cryptonote_core/service_node_list.cpp

```diff
@@ -312,7 +320,114 @@ namespace service_nodes
 
 
     m_rollback_events.push_back(std::unique_ptr<rollback_event>(new rollback_change(block_height, key, iter->second)));
+    uint64_t swarm_id = iter->second.swarm_id;
     m_service_nodes_infos.erase(iter);
+    check_and_process_swarm_decomission(swarm_id);
+  }
+
+  void service_node_list::check_and_process_swarm_decomission(uint64_t swarm_id)
+  {
+    size_t count = 0;
+    for (const auto& entry : m_service_nodes_infos)
+      if (entry.second.swarm_id == swarm_id)
+        count++;
+    if (count == MIN_SWARM_SIZE - 1)
+    {
+      // XXX XXX XXX XXX
+      // All nodes should fetch data from this swarm at this point only.
+      // The registered nodes will eventually disappear and after this point,
+      // it is already considered gone. It only exists to retrieve data from.
```

Comment:
Thoughts on stealing nodes 1 by 1 from the largest swarm until you become >= min_swarm_count in an attempt to backup the data and preserve the swarm ID? If insufficient, then merge.

Reply:
I don't like it because it's complicated and difficult to evaluate the bandwidth savings. If the numbers are picked right, this critically low case shouldn't happen too often.

Could be worth it though, haven't done any analysis on this. I'd recommend implementing this stupidly simple version, then later we can make incremental improvements to the swarm selection algorithm if it's a problem.